src/duk_debugger.c

```diff
@@ -847,6 +847,51 @@ DUK_INTERNAL void duk_debug_send_status(duk_hthread *thr) {
 	duk_debug_write_eom(thr);
 }
 
+DUK_INTERNAL void duk_debug_send_throw(duk_hthread *thr, duk_bool_t fatal) {
+	/*
+	 * NFY <int: 5> <int: fatal> <str: msg> <str: filename> <int: linenumber> EOM
+	 */
+
+	duk_context *ctx = (duk_context *)thr;
+	duk_activation *act;
+	duk_uint32_t pc;
+
+	DUK_ASSERT(thr->valstack_top > thr->valstack);  /* ... [err] */
+
+	duk_debug_write_notify(thr, DUK_DBG_CMD_THROW);
+	duk_debug_write_int(thr, fatal);
+
+	/* Report thrown value to client coerced to string */
+	duk_dup(ctx, -1);
+	duk_safe_to_string(ctx, -1);
+	duk_debug_write_hstring(thr, duk_require_hstring(ctx, -1));
+	duk_pop(ctx);
+
+	if (duk_is_error(ctx, -1)) {
+		/* Error instance, use augmented error data directly */
+		duk_get_prop_stridx(ctx, -1, DUK_STRIDX_FILE_NAME);
+		duk_safe_to_string(ctx, -1);
+		duk_debug_write_hstring(thr, duk_require_hstring(ctx, -1));
+		duk_get_prop_stridx(ctx, -2, DUK_STRIDX_LINE_NUMBER);
+		duk_debug_write_uint(thr, duk_require_uint(ctx, -1));
```

Comment:
Hmm, regarding `duk_require_hstring()` here, it'd be cleaner to use `duk_get_hstring()` (here and in existing debugger codebase) and ensure `duk_debug_write_hstring()` wrote an empty string on a NULL input.

This is the case because throwing an error is never correct behavior here. So the best choices would be IMO: (1) `duk_get_hstring()` + assert, or (2) `duk_get_hstring()` and silently tolerating a NULL.

No need to fix these in this pull - but I'll add a note to clean this up later on.

The `duk_require_uint()` issue still remains, and it's an actual potential problem. You should be able to reproduce the issue by constructing an error like:

```
duk> e = new Error('foo'); Object.defineProperty(e, 'lineNumber', {value:"foo"})
= Error: foo
duk> e.lineNumber
= foo
```